tests/test_nblist.py

```diff
@@ -376,7 +380,114 @@ def test_nblist_max_interactions(block_size, tiles):
+            )
+            box += np.diag([0.1, 0.1, 0.1])  # remove any possible clashes
+        host_config = HostConfig(host_sys, host_conf, box, num_water_atoms)
+    elif host_name == "solvent":
+        host_sys, host_conf, box, _ = build_water_system(4.0, forcefield.water_ff, mols=[mol_a, mol_b])
+        box += np.diag([0.1, 0.1, 0.1])  # remove any possible clashes
+        host_config = HostConfig(host_sys, host_conf, box, host_conf.shape[0])
+    else:
+        assert 0, "Invalid host name"
+
+    st = SingleTopology(mol_a, mol_b, core, forcefield)
+    host = setup_optimized_host(st, host_config)
+
+    lambda_grid = np.array([0.0])
+
+    initial_state = setup_initial_states(st, host, DEFAULT_TEMP, lambda_grid, seed=2024, min_cutoff=None)[0]
+
+    return st, host, host_name, initial_state
+
+
+def compute_mean_block_occupancy(
+    nblist, frame: NDArray, box: NDArray, cutoff: float, column_block: int = 32, row_block: int = 32
+) -> float:
+    assert column_block == 32
+    assert column_block == row_block
+    N = frame.shape[0]
+    assert nblist.get_num_row_idxs() == N
+    ixn_list = nblist.get_nblist(frame, box, cutoff)
+    block_occupancies = []
+    max_ixn_per_block = column_block * row_block * ((N + row_block - 1) // row_block)
+    max_ixn_per_block = min(N, max_ixn_per_block)
+
+    for i, ixn_block in enumerate(ixn_list):
+        assert max_ixn_per_block >= len(ixn_block)
+        block_occupancies.append(len(ixn_block) / max_ixn_per_block)
+        # Each interaction group can hold column_block fewer ixns as it is upper triangular
+        max_ixn_per_block -= column_block
+    return float(np.mean([block_occupancies]))
+
+
+@pytest.mark.parametrize("cutoff", [0.9, 1.0, 1.2])
+@pytest.mark.parametrize("precision", [np.float32, np.float64])
+def test_nblist_occupancy_dhfr(cutoff, precision):
+    _, _, coords, box = setup_dhfr()
+
+    if precision == np.float32:
+        nblist = custom_ops.Neighborlist_f32(coords.shape[0])
+    else:
+        nblist = custom_ops.Neighborlist_f64(coords.shape[0])
+    occupancy = compute_mean_block_occupancy(nblist, coords, box, cutoff)
+    assert occupancy > 0.10
+    print(f"DHFR NBList Occupancy with Cutoff {cutoff}: {occupancy * 100.0:.3g}%")
+
+
+@pytest.mark.parametrize("frames", [10])
+@pytest.mark.parametrize("precision", [np.float32])
+def test_nblist_occupancy(hif2a_single_topology_leg, frames, precision):
+    cutoff = 1.2
+    st, host, host_name, initial_state = hif2a_single_topology_leg
+
+    md_params = MDParams(n_eq_steps=0, steps_per_frame=100, n_frames=frames, seed=2024)
+
+    traj = sample(initial_state, md_params, max_buffer_frames=md_params.n_frames)
+
+    if precision == np.float32:
+        nblist = custom_ops.Neighborlist_f32(initial_state.x0.shape[0])
+    else:
+        nblist = custom_ops.Neighborlist_f64(initial_state.x0.shape[0])
+    occupancies = []
+    occupancy = compute_mean_block_occupancy(nblist, initial_state.x0, initial_state.box0, cutoff)
```

Comment:
is `initial_state.x0` equilibrated? (equilibrated tile densities will be different from non-equilibrated tile densities)

Reply:
It will have a pre-equilibrated (1000 steps with Barostat, from `pre_equilibrate_host`) but is not equilibrated. If you run this out longer you can see a significant shift in the nblist density.